gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,1268 @@
+      .ttl(25, TimeUnit.SECONDS)
+      .build();
+  public static final DnsRecord AAAA_RECORD_ZONE1 = DnsRecord
+      .builder("www." + ZONE1.dnsName(), DnsRecord.Type.AAAA)
+      .records(ImmutableList.of("ed:ed:12:aa:36:3:3:105"))
+      .ttl(25, TimeUnit.SECONDS)
+      .build();
+  public static final ChangeRequest CHANGE_ADD_ZONE1 = ChangeRequest.builder()
+      .add(A_RECORD_ZONE1)
+      .add(AAAA_RECORD_ZONE1)
+      .build();
+  public static final ChangeRequest CHANGE_DELETE_ZONE1 = ChangeRequest.builder()
+      .delete(A_RECORD_ZONE1)
+      .delete(AAAA_RECORD_ZONE1)
+      .build();
+
+  public static void purge() {
+    Page<Zone> zones = DNS.listZones();
+    Iterator<Zone> zoneIterator = zones.iterateAll();
+    while (zoneIterator.hasNext()) {
+      Zone zone = zoneIterator.next();
+      List<DnsRecord> toDelete = new LinkedList<>();
+      if (zone.name().startsWith(PREFIX)) {
+        Iterator<DnsRecord> dnsRecordIterator = zone.listDnsRecords().iterateAll();
+        while (dnsRecordIterator.hasNext()) {
+          DnsRecord record = dnsRecordIterator.next();
+          if (!ImmutableList.of(DnsRecord.Type.NS, DnsRecord.Type.SOA).contains(record.type())) {
+            toDelete.add(record);
+          }
+        }
+        if (!toDelete.isEmpty()) {
+          zone.applyChangeRequest(ChangeRequest.builder().deletions(toDelete).build());
+        }
+        zone.delete();
+      }
+    }
+  }
+
+  private static List<Zone> filter(Iterator<Zone> iterator) {
+    List<Zone> result = new LinkedList<>();
+    while (iterator.hasNext()) {
+      Zone zone = iterator.next();
+      if (zone.name().startsWith(PREFIX)) {
+        result.add(zone);
+      }
+    }
+    return result;
+  }
+
+  @BeforeClass
+  public static void before() {
+    purge();
+  }
+
+  @AfterClass
+  public static void after() {
+    purge();
+  }
+
+  @Test
+  public void testCreateValidZone() {
+    Zone created = DNS.create(ZONE1);
+    assertEquals(ZONE1.description(), created.description());
+    assertEquals(ZONE1.dnsName(), created.dnsName());
+    assertEquals(ZONE1.name(), created.name());
+    assertNotNull(created.creationTimeMillis());
+    assertNotNull(created.nameServers());
+    assertNull(created.nameServerSet());
+    assertNotNull(created.id());
+    Zone retrieved = DNS.getZone(ZONE1.name());
+    assertEquals(created, retrieved);
+    created = DNS.create(ZONE_EMPTY_DESCRIPTION);
+    assertEquals(ZONE_EMPTY_DESCRIPTION.description(), created.description());
+    assertEquals(ZONE_EMPTY_DESCRIPTION.dnsName(), created.dnsName());
+    assertEquals(ZONE_EMPTY_DESCRIPTION.name(), created.name());
+    assertNotNull(created.creationTimeMillis());
+    assertNotNull(created.nameServers());
+    assertNull(created.nameServerSet());
+    assertNotNull(created.id());
+    retrieved = DNS.getZone(ZONE_EMPTY_DESCRIPTION.name());
+    assertEquals(created, retrieved);
+  }
+
+  @After
+  public void tearDown() {
+    purge();
+  }
```

Comment:
Instead of calling `purge()` after each test wouldn't it be better to selectively delete what was created for the test? 

I would still keep it in `@afterClass` to clean up in case some test failed